3rdparty/mbedtls/mbedtls/library/x509write_crt.c

```diff
@@ -308,7 +308,7 @@ int mbedtls_x509write_crt_der( mbedtls_x509write_cert *ctx, unsigned char *buf,
     unsigned char *c, *c2;
     unsigned char hash[64];
     unsigned char sig[MBEDTLS_MPI_MAX_SIZE];
-    unsigned char tmp_buf[2048];
+    unsigned char tmp_buf[2048*4];
```

Comment:
Oh I remember you talking about this a few months ago, that mbedtls doesn't support large enough asn1 tags to hold a quote. That's unfortunate! It would mean that a user that wants to replace the mbedtls library at the upcoming oecrypto interface with a different version of mbedtls, they would need to make sure their mbedtls library is patched like this. I'm fine with that, as long as we make it clear to the user replacing this! :)

Reply:
Yes. That's true.
I am in a process of trying to get those mbedtls buffer size changes to upstream.
(openssl does not have this limitation)

Reply:
Nice!

Reply:
We have avoided editing 3rd party code as much as possible. Can we instead make these changes via a patch that gets applied as part of the build system? That way these changes will not be lost when we upgrade mbedtls.

Reply:
This is a very good point. I will make it's done as suggested.

Reply:
Even better would be to NOT copy 3rd party code in the first place, but instead use a git submodule, and then you can submit into a branch (fork) of mbedtls that you depend on, until the branch gets merged.

Reply:
I opened a mbetls issue https://github.com/ARMmbed/mbedtls/issues/2631 for tracking this mbedtls limitation. And an PR with a clean in-place solution (https://github.com/ARMmbed/mbedtls/pull/2632) was posted an ARM developer shortly after my posting. 

With that said,  once we pick up the mbedtls release with above PR, we do not need to maintain the private mbedtls changes I included in this PR.